### ISSUE TITLE
`def __init__(self: Self)` causes `reportInvalidTypeVarUse`

### ISSUE THREAD

**Issue by minacle** [microsoft/pyright#7986]
I am using Pylance v2024.5.102 (pre-release) extension through Visual Studio Code 1.89.1.

The following code can trigger this message.

```python
from __future__ import annotations

from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from typing_extensions import Self


class A:

    def __init__(self: Self, /) -> None:
        pass
```

The Python version is 3.11.9.

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

The `reportInvalidTypeVarUse` check is intended to detect cases where a type variable is employed where it is not needed. Developers who are not familiar with generics in the Python type system often use type variables in places where they are unneeded. This diagnostic is intended to inform you of this case and guide you to the proper use of type variables.

In your code sample, you're specifying a type of `Self` for variable `self` when it already has that type. This type annotation is completely redundant and unnecessary. You can simply delete the annotation.

If you don't want to be told about unnecessary or invalid use of type variables, you can disable the `reportInvalidTypeVarUse` check.
